2010-10-12  Michael Tautschnig  <mt@debian.org>

	* fai, make-fai-nfsroot: Use df -P for guaranteed one-line-is-one-mount output
		(thanks michael log <mmlogin@gmail.com> for the patch).
Index: trunk/bin/fai
===================================================================
--- trunk.orig/bin/fai
+++ trunk/bin/fai	
@@ -225,7 +225,7 @@
     FAI_ROOT=$(cd $FAI_ROOT;pwd)
 
     # check if target directory is mounted with bad options
-    fs=$(df $FAI_ROOT | tail -1 | awk '{print $6}')
+    fs=$(df -P $FAI_ROOT | tail -1 | awk '{print $6}')
     if mount | grep "on $fs " |  awk '{print $6}' | egrep -q "nosuid|nodev"; then
 	echo "Target directory is mounted using nosuid or nodev. Aborting"
 	exit 5
Index: trunk/bin/make-fai-nfsroot
===================================================================
--- trunk.orig/bin/make-fai-nfsroot
+++ trunk/bin/make-fai-nfsroot	
@@ -234,7 +234,7 @@
 call_debootstrap() {
     
     # check if NFSROOT directory is mounted with bad options
-    fs=$(df $NFSROOT | tail -1 | awk '{print $6}')
+    fs=$(df -P $NFSROOT | tail -1 | awk '{print $6}')
     if mount | grep "on $fs " |  awk '{print $6}' | egrep -q "nosuid|nodev"; then
         die 1 "NFSROOT directory $NFSROOT is mounted using nosuid or nodev. Aborting"
     fi
